packages/ng-query/src/lib/operators/filter-success.operator.ts

```diff
@@ -0,0 +1,10 @@
+import { filter, Observable } from 'rxjs';
+import { NgQueryObserverResult } from '../types';
+
+export function filterSuccess() {
+  return function <T>(
+    source: Observable<NgQueryObserverResult<T>>
+  ): Observable<NgQueryObserverResult<T>> {
+    return source.pipe(filter((result) => result.status === 'success'));
```

Comment:
We'll want to infer the `data` has not nullable in this case—checkout filter with type guards.

Reply:
I have added type guard but not sure how would I do the same for the filterError operator

Reply:
Same, but `data` should be `null` and not `T`